lib/kalibro_client/entities/processor/metric_collector_details.rb

```diff
@@ -33,8 +33,7 @@ def find_metric_by_name(name)
         end
 
         def find_metric_by_code(metric_code)
-          metric = self.supported_metrics.find {|code, metric| metric.code == metric_code}
-          metric.nil? ? nil : metric.last
+          @supported_metrics[metric_code]
```

Comment:
I would guess the performance issues are more likely related to figuring out whether each collector is available than actually searching for them in an array. There are not even 50 metrics after all, even a linear search should be instant.